Get-NetView.psm1

```diff
@@ -971,6 +971,64 @@ function MellanoxFirmwareInfo {
     ExecCommandsAsync -OutDir $dir -File $file -Commands $cmds
 } # MellanoxFirmwareInfo()
 
+function MellanoxWinOFTool{
+    [CmdletBinding()]
+    Param(
+        [parameter(Mandatory=$false)] [String] $NicName,
+        [parameter(Mandatory=$true)] [String] $OutDir
+    )
+
+    $dir = $OutDir
+
+    $toolName = "mlxtool.exe"
+    $toolPath = "$env:ProgramFiles\Mellanox\MLNX_VPI\Tools\$toolName"
+    $toolCmd = "&""$toolPath"""
+
+    $hardwareInfo = Get-NetAdapterHardwareInfo -Name $NicName
+    $deviceLocation = "$($hardwareInfo.bus)`_$($hardwareInfo.device)`_$($hardwareInfo.function)"
+
+    $showArgs = "ports",
+                "devices",
+                "tc-bw",
+                "vxlan",
+                "ecn config",
+                "packet-filter",
+                "qos",
+                "regkeys all miniport",
+                "regkeys all bus",
+                "nd connections",
+                "ndk connections",
+                "perfstats ""$NicName"" showall",
+                "driverparams",
+                "selfhealing port"
+
+    $dbgArgs = "oid-stats-ext",
+               "cmd-stats-ext",
+               "resources",
+               "pkeys",
+               "ipoib-ep",
+               "get-state",
+               "rfd-profiling ""$NicName"" dump",
+               "pddrinfo",
+               "dump-me-now",
+               "eq-data ""$deviceLocation""",
+               "dma-cached-stats ""$deviceLocation"""
+
+    [String []] $runCmds = @()
+    foreach ($arg in $showArgs)
+    {
+        $runCmds += $toolCmd + " show " + $arg
+    }
+
+    foreach ($arg in $dbgArgs)
+    {
+        $runCmds += $toolCmd + " dbg " + $arg
+    }
+    $file = "mlxtoolOutput.txt"
+    ExecCommandsAsync -OutDir $dir -File $file -Commands $runCmds
```

Comment:
For consistent style, write out the full commands instead of generating them. Can still use the `$toolCmd` variable. Also, you may want to split these commands into multiple files to make them easier to find. I would do mlxTool.show.txt and mlxTool.dbg.txt at least, but you may want to do more granular.

Reply:
Done. A single output file is sufficient for now.